File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        s = ""
        res = ""
        n = len(num)
        for i in range(1, n-1, 1):
            if num[i-1] == num[i] and num[i] == num[i+1]:
                s = num[i-1] + num[i] + num[i+1]
                res = s if len(res) == 0 else s if ord(s[0]) > ord(res[0]) else res
        return res
        

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string res = "";
        int n = num.length();
        for (int i = 1; i < n - 1; i++) {
            if (num[i - 1] == num[i] && num[i] == num[i + 1]) {
                string s = num.substr(i - 1, 3); // Extract substring of length 3 around index i
                res = (res.empty()) ? s : (s > res) ? s : res;
            }
        }
        return res;
    }
};